presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java

```diff
@@ -365,6 +366,10 @@ private void runCommit(long transactionId, HandleConsumer callback)
                 return;
             }
             catch (DBIException e) {
+                if (isTransactionCacheFullError(e)) {
+                    throw metadataError(e, "A possible cause of this problem is inserting a table that has too many columns");
```

Comment:
Let's change the message to the one saying that transaction size is too big. Also, it's not just the number of columns, but also number of rows (shards) and sizes of columns (like long varchars).